query/src/main/java/org/infinispan/query/impl/LifecycleManager.java

```diff
@@ -475,4 +485,31 @@ public void cacheManagerStarting(GlobalComponentRegistry gcr, GlobalConfiguratio
    public void cacheManagerStopped(GlobalComponentRegistry gcr) {
       mbeanServer = null;
    }
+
+   /**
+    * Sets {@link BooleanQuery#setMaxClauseCount} according to the value of {@link #MAX_BOOLEAN_CLAUSES_SYS_PROP} system
+    * property. This is executed only once, when first indexed cache is started.
+    */
+   private void setBooleanQueryMaxClauseCount() {
+      if (!maxBooleanClausesWasSet) {
```

Comment:
I would do 
```java 
if(maxBooleanClausesWasSet) return;
```
And then the bloc

Reply:
I do not like returning from the middle of the method. Unless the method is long and boring, then I do return abruptly.

Reply:
is not the middle of the method, is the beginning! as there are additional if statements inside, imho flattens it helps to understand, but ok, it's simple enough like this